src/libcore/clone.rs

```diff
@@ -46,14 +46,39 @@
 
 use marker::Sized;
 
-/// A common trait for cloning an object.
+/// A common trait for cloning an object. Differs from `Copy` in that you can
+/// define `Clone` to run arbitrary code, while you are not allowed to override
+/// the implementation of `Copy` that only does a `memcpy`.
```

Comment:
Okay, GitHub is being weird, and when I refreshed, my comments were gone, so trying again. Sorry if this is a duplicate.

First, this summary line is a bit long. We try to keep the very first line a sentence, and while what was here was a bit... sparse, the summary line should still be shorter than this.

Second, the `Copy`/`Clone` distinction isn't really about overriding, exactly. Or rather, to me, the not-overriding is the _implementation_ rather than the important bit. The important bit is that `Copy` is extremely inexpensive and is implicit. `Clone` may or may not be expensive, and is always explicit. The mechanism to ensure this is to not let you implement the details of a `Copy`.

I also have small concerns about putting stuff about `Copy`'s semantics right here in the summary; while they won't really be changing.... idk. Hm.